src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -766,7 +774,7 @@ protected override TypeWithAnnotations InferTypeOfVarVariable(DiagnosticBag diag
                         _nodeBinder.BindConstructorInitializer(initializer, diagnostics);
                         break;
                     case SyntaxKind.ArgumentList:
-                        var invocation = (ConstructorInitializerSyntax)_nodeToBind.Parent;
+                        var invocation = (ConstructorInitializerSyntax)_nodeToBind.Parent!;
```

Comment:
>Parent! [](start = 83, length = 7)

How do we know?

Reply:
https://github.com/dotnet/roslyn/blob/887406115e6e2dcee2d52a009c8f5bc5fa67844f/src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs#L750